cypress/page/personal-Form.page.ts

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable require-jsdoc */
+class PersonalFormPage {
+  private toolsQaFormURL: string;
+  private firstName: string;
+  private lastName: string;
+  private email: string;
+  private gender: string;
+  private mobileNumber: string;
+  private hobbies: string;
+  private currentAddresss: string;
+  private submit: string;
+  private submitting: string;
+  private state: string;
+  private stateOptions: string;
+  private city: string;
+
+  constructor() {
+    this.toolsQaFormURL = "https://demoqa.com/automation-practice-form";
+    this.firstName = "#firstName";
+    this.lastName = "#lastName";
+    this.email = "#userEmail";
+    this.gender = "#genterWrapper .col-md-9";
+    this.mobileNumber = "#userNumber";
+    this.hobbies = "#hobbiesWrapper .col-md-9";
+    this.currentAddresss = "#currentAddress";
+    this.state = "#state";
+    this.stateOptions = ".css-26l3qy-menu";
+    this.city = "#city";
+    this.submit = "#userForm";
+    this.submitting = ".modal-title";
+  }
+
+  public completeAllTheProcess(): any {
+    return cy.get(this.submitting);
+  }
+
+  private findGenderByName(name: string): any {
+    return cy.get(this.gender).filter(`:contains("${name}")`);
+  }
+
+  private findHobbiesByName(hobbies: string[]): any {
+    hobbies.forEach((hobbie) => {
+      cy.get(this.hobbies).find(`label:contains("${hobbie}")`).click();
```

Comment:
and here, instead of label:contains, I would use .find(`[value="${hobby}"]`) or filter. And then I would use check, instead of click, but this is to give more context about the element we are interacting with.

Reply:
Holmes in this case the Value for each Hobby is a number [value='1'].... and it means that its not possible to find using this selector.